T1174

secretrobotron opened the pull request.

secretrobotron committed:
[t1174] iframe-less editors working

secretrobotron committed:
[t1174] removed redundant/old/useless files

secretrobotron committed:
[t1174] slightly better approach, fully working

dseif commented on css/butter.ui.deprecated.css:
```diff
@@ -1495,3 +1499,71 @@ input.form-error {
+
+#butter-timeline .butter-default-editor {
+
+}
```
why empty?

secretrobotron replied:
Hahahahahaahah noooo reason at all.

dseif commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6bfff6e01cc5473/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6bfff6e01cc5473/output.txt

Total script time: 0.30 mins

dseif commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/31f11dc56c23ec1/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/31f11dc56c23ec1/output.txt

Total script time: 0.95 mins

Server is running on http://pr388.butyr.org/templates/test.html

dseif commented on editors/default.js:
```diff
@@ -0,0 +1,259 @@
+//       sendData( false );
+//     });
+//   }, false );
+// })();
```
This is missing the default stuff stuff that I did in the default editor and checkboxes no longer work ( check one, change to a different trackevent, then change back and it doesn't keep its state ).

Also I get an error when changing the start end times to various numbers ( try deleting it completely, changing to a different field, and going back ).

I get
    Uncaught Error: Invalid time format

secretrobotron replied:
My bad. Also, this commented code shouldn't be here. Lemme fix.

dseif commented on editors/default.html:
```diff
@@ -0,0 +1,8 @@
+      If a copy of the MIT license was not distributed with this file, you can
+      obtain one at http://www.mozillapopcorn.org/butter-license.txt -->
+
+<div class="butter-editor">
```
So to create a template now all you need to do is a have a container element with an id of butter-editor instead of an iframe?

secretrobotron replied:
That's right!

dseif replied:
That's pretty awesome

secretrobotron committed:
[t1174] ported old checkboxes/defaults/errors

secretrobotron committed:
[t1174] deleted old code

secretrobotron committed:
[t1174] added tests and fixed up minor bugs

dseif commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/16ce55bd23216cd/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/16ce55bd23216cd/output.txt

Total script time: 0.16 mins

dseif commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/be5f440a3846afc/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/be5f440a3846afc/output.txt

Total script time: 0.14 mins

secretrobotron committed:
[t1174] fixing up - WIP

secretrobotron committed:
[t1174] added more spacing to editor area

secretrobotron committed:
[t1174] regression fixes

1. Blinking in default editor
2. applyDefaults conditional in trackevent
3. Better test data in test template

secretrobotron committed:
[t1174] removed console.log

secretrobotron committed:
[t1174] bug fixes and isRegistered

1. Added Editor.isRegistered to check for editor types
2. Fixed bug in reading options from manifest to compare with textboxes in default editor

secretrobotron committed:
[t1174] src/editor/module documentation

secretrobotron committed:
[t1174] documented src/editor/editor.js

secretrobotron committed:
[t1174] documented editors/default.js

secretrobotron committed:
[t1174] fixed lint errors from documentation update

secretrobotron committed:
[t1174] proper UI layout for editor area

1. Created editor-area.html layout
2. Exposed Area on butter.ui
3. Editor module creates its own UI conditions (adds spacing, adds Area, etc.)
4. Changed 'editor-area' to 'butter-editor-area' in CSS

secretrobotron committed:
[t1174] toggler fixes

1. Made Toggler class more accessible so that other components can use it
2. Added toggler construct to editor area
3. Fixed toggler css

secretrobotron committed:
[t1174] wip - editor layout load support

secretrobotron committed:
[t1174] final touches & cleanup before pullrequest

1. Removed googlemap editor from this commit (will put in new branch)
2. Moved default editor into src/
3. Removed editors from configs
4. Removed include folder (not needed anymore)
5. Exposed Editor and baseDir to Editor authors
6. Refactored lots of default-editor code to be part of BaseEditor (better future support)
7. Fixed a couple of bugs from empty trackEvent or bad close() calls, etc.

secretrobotron committed:
[t1174] fixed bug when ui is off for editors

secretrobotron committed:
[t1174] lint cleanup after rebase

secretrobotron committed:
[t1174] removed trailing comma in build.js

humphd commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5c21ebabf36ffa3/output.txt

humphd commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/1d6bc59bfe312bc/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5c21ebabf36ffa3/output.txt

Total script time: 0.82 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/1d6bc59bfe312bc/output.txt

Total script time: 1.08 mins

Server is running on http://pr388.butyr.org/templates/test.html

humphd commented on css/butter.ui.deprecated.css:
```diff
@@ -1421,3 +1415,102 @@ input.form-error {
+  right: 0;
+  width: 200px;
+  border: 1px solid #000;
+  background-image: url( "../resources/bg.jpg" );
```
Why not a dataurl like the other icons we use?

secretrobotron replied:
Are the icons data-uri? I know that we reference `resources/bg.jpg` a few times in this file, so I followed suit.

humphd commented on src/editor/default.js:
```diff
@@ -0,0 +1,129 @@
+ * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
+
+define( [ "text!./default.html",
+          "editor/editor", "util/lang" ],
```
Move "util/lang" to new line so it's more obvious it's there

humphd commented on src/editor/editor.js:
```diff
@@ -2,149 +2,426 @@
-  var DEFAULT_DIMS = [ 400, 400 ];
+  var __editors = {},
+      __defaultLayouts = LangUtils.domFragment( DEFAULT_LAYOUT_SNIPPETS ),
+      __safeKeyUpKeys = [ 37, 38, 39, 40, 8, 27 ];
```
Don't use magic numbers, define variables so it's obvious what these are.

secretrobotron committed:
[t1174] review changes

1. Added util/keys.js for keyboard map
2. Styling fixes
3. Used key map in default editor

secretrobotron committed:
[t1174] review changes

1. Double-click trackevent automatically opens editor
2. Added better state control to toggler and fixed editor module's use of it
3. Added ignore key list to BaseEditor.createPropertiesFromManifest

secretrobotron committed:
[t1174] review changes

1. Fixed small bug in trackevent where manifestOptions aren't defined as expected
2. Removed files that snuck in from another merge/rebase

secretrobotron committed:
[t1174] lint cleanup

secretrobotron committed:
[t1174] tests

1. Editor exposed on Butter in main.js
2. Tests for Editor
3. Small bug fixes found (usually layout work) from testing

dseif commented on src/editor/default.js:
```diff
@@ -0,0 +1,128 @@
+ * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
+
+define( [ "text!./default.html", "editor/editor", "util/lang" ],
+  function( LAYOUT_SRC, Editor, LangUtils ) {
```
Do we need to include util/lang, doesn't look like its used any where

dseif commented on src/editor/editor.js:
```diff
@@ -2,149 +2,449 @@
+          itemLabel = "Start (seconds)";
+        } else if ( itemLabel === "Out" ) {
+          itemLabel = "End (seconds)";
+        }
```
There is a ticket in Popcorn to get rid of In and Out and make them Start and End, so we can probably get rid of this and just do a submodule update

secretrobotron committed:
[t1174] review fixes

1. Toggler bug in editor (state not being correctly applied)
2. Removed LangUtil from default editor js

secretrobotron committed:
[t1174] some merge/rebase confusion resolved

dseif commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5d74aeaa92d6f37/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5d74aeaa92d6f37/output.txt

Total script time: 0.66 mins

Server is running on http://pr388.butyr.org/templates/test.html
